paddle/fluid/pybind/eager.cc

```diff
@@ -233,298 +443,223 @@ int EagerTensorInit(PyObject* self, PyObject* args, PyObject* kwds) {
+                /* false means not init by egr tensor*/ false);
+            return 0;
+          }
+        } else if (kw_dtype != NULL &&
+                   PyObject_IsInstance(kw_dtype, reinterpret_cast<PyObject*>(
+                                                     g_vartype_pytype))) {
+          VLOG(6) << "Calling case2's initializer";
+
+          PADDLE_ENFORCE_NOT_NULL(
+              kw_dims,
+              paddle::platform::errors::Fatal(
+                  "Calling __init__ of Eager Tensor with NULL dims is "
+                  "forbidden. Please check your code and make sure you new a "
+                  "dims before calling this constructor."));
+
+          PADDLE_ENFORCE_NOT_NULL(
+              kw_name,
+              paddle::platform::errors::Fatal(
+                  "Calling __init__ of Eager Tensor with NULL name is "
+                  "forbidden. Please check your code and make sure you new a "
+                  "name before calling this constructor."));
+
+          PADDLE_ENFORCE_NOT_NULL(
+              kw_dtype,
+              paddle::platform::errors::Fatal(
+                  "Calling __init__ of Eager Tensor with NULL dtype is "
+                  "forbidden. Please check your code and make sure you new a "
+                  "dtype before calling this constructor."));
+
+          PADDLE_ENFORCE_NOT_NULL(
+              kw_persistable,
+              paddle::platform::errors::Fatal(
+                  "Calling __init__ of Eager Tensor with NULL persistable is "
+                  "forbidden. Please check your code and make sure you new a "
+                  "persistable before calling this constructor."));
+
+          paddle::framework::proto::VarType::Type dtype =
+              CastPyArg2ProtoType(kw_dtype, 0);
+          std::vector<int> dims = CastPyArg2VectorOfInt(kw_dims, 0);
+
+          std::string act_name = "";
+          if (kw_name == Py_None) {
+            act_name = egr::Controller::Instance().GenerateUniqueName(
+                "generated_tensor");
+          } else {
+            act_name = CastPyArg2AttrString(kw_name, 0);
+          }
+
+          paddle::framework::proto::VarType::Type var_type =
+              CastPyArg2ProtoType(kw_type, 0);
+          bool persistable = CastPyArg2AttrBoolean(kw_persistable, 0);
+
+          EmptyEagerTensorInitializer(
+              py_tensor_ptr, act_name,
+              egr::Controller::Instance().GetExpectedPlace(), persistable,
+              /* stop_gradient */ true, dtype, dims, var_type);
+
+          return 0;
+        } else {
+          PADDLE_THROW(platform::errors::InvalidArgument(
+              "We support construct tensor from numpy value or tensor "
+              "with python args and kwargs by this initializer, "
+              "please check your input first and make sure you are on the "
+              "right way."));
+        }
       }
-      return 0;
     }
-    case (Py_ssize_t)2: {
+    case (Py_ssize_t)1:
```

Comment:
why this? empty case?

Reply:
Because case 1, case 2, case 3 can use the same logic. But switch case not used now.